models/note.go

```diff
@@ -22,16 +23,33 @@ var categoryStrings = [...]string{
 	"predictions",
 }
 
-func (category Category) String() string {
-	if category < MARGINALIA || category > PREDICTIONS {
-		return "Unknown"
+var UnDeserializeableCategoryStringError = errors.New("String does not correspond to a Note Category")
+
+func DeserializeCategory(input string) (Category, error) {
+	for i := 0; i < len(categoryStrings); i++ {
+		if input == categoryStrings[i] {
+			return Category(i), nil
+		}
 	}
+	return MARGINALIA, UnDeserializeableCategoryStringError
+}
 
+func (category Category) String() string {
 	return categoryStrings[category]
```

Comment:
what does this do when `category` is out of bounds?

Reply:
When you try to convert it to a type `Category` it will be either a cast or a compiler error.

Reply:
oh, wait. are you saying the out of bounds error is impossible?

Reply:
like there's no way to pass an invalid `category` value into this function?

Reply:
I was wrong. I just tested it with this code and you can cast any number to Category. So i've added the check which returns "Unknown"